ppa/archive/tests/test_import_util.py

```diff
@@ -269,6 +268,26 @@ def test_rsync_data(self, mocksubprocess):
         assert "ppa_hathi_pathlist" in cmd_args[-3]
 
 
+def test_hathiimporter_init(tmp_path_factory):
+    # no rsync output, no output dir
+    htimporter = HathiImporter(["hvd.1234", "nyp.334455"])
+    assert htimporter.rsync_output is False
+    assert htimporter.output_dir is None
+
+    # rsync output requested with no output dir
+    with pytest.raises(ValueError, match="output_dir is required"):
+        HathiImporter(rsync_output=True)
+
+    # rsync output requested with non-existent output dir
+    with pytest.raises(ValueError, match="not an existing directory"):
+        # rsync output requested with invalid output dir
+        HathiImporter(rsync_output=True, output_dir="/tmp/foo/bar")
```

Comment:
So CodeFactor won't complain (there's no reason `tmp` needs to be mentioned)
```suggestion
        HathiImporter(rsync_output=True, output_dir="/foo/bar/baz")
```